src/lib/extension-load-queue.ts

```diff
@@ -0,0 +1,86 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type BrowserWindow = any;
+
+export type ControllerLoadedCallback<T> = (controller: T) => void;
+
+export const QUEUE_SYMBOL: unique symbol = Symbol.for('queue');
+
+export type ScriptStore<T> = ControllerLoadedCallback<T>[] | null;
```

Comment:
Wouldn't throwing an error ("This should only be used client-side") make more sense here though? In particular, it would allow to keep the API a bit more consistent without that `null`